.github/workflows/test.yml

```diff
@@ -31,148 +31,16 @@ jobs:
-              "itest-self_sent_txn": ["self-hosted", "linux", "x64", "xlarge"],
-              "itest-verifreg": ["self-hosted", "linux", "x64", "xlarge"],
-              "multicore-sdr": ["self-hosted", "linux", "x64", "xlarge"],
-              "unit-node": ["self-hosted", "linux", "x64", "xlarge"]
-            }
-          # A list of test groups that require Proof Parameters to be fetched
-          parameters: |
-            [
-              "conformance",
-              "itest-api",
-              "itest-direct_data_onboard_verified",
-              "itest-direct_data_onboard",
-              "itest-manual_onboarding",
-              "itest-niporep_manual",
-              "itest-net",
-              "itest-path_detach_redeclare",
-              "itest-sealing_resources",
-              "itest-sector_import_full",
-              "itest-sector_import_simple",
-              "itest-sector_pledge",
-              "itest-sector_unseal",
-              "itest-wdpost_no_miner_storage",
-              "itest-wdpost_worker_config",
-              "itest-wdpost",
-              "itest-worker",
-              "multicore-sdr",
-              "unit-cli",
-              "unit-storage"
-            ]
         run: |
-          # Create a list of integration test groups
-          itests="$(
-            find ./itests -name "*_test.go" | \
-              jq -R '{
-                "name": "itest-\(. | split("/") | .[2] | sub("_test.go$";""))",
-                "packages": [.]
-              }' | jq -s
-            )"
-
-          # Create a list of packages that are covered by the integration and unit tests
-          packages="$(jq -n --argjson utests "$utests" '$utests | map(.packages) | flatten | . + ["./itests/..."]')"
-
-          # Create a new group for the unit tests that are not yet covered
-          rest="$(
-            find . -name "*_test.go" | cut -d/ -f2 | sort | uniq | \
-              jq -R '"./\(.)/..."' | \
-              jq -s --argjson p "$packages" '{"name": "unit-rest", "packages": (. - $p)}'
-          )"
-
-          # Combine the groups for integration tests, unit tests, the new unit-rest group, and the other tests
-          groups="$(jq -n --argjson i "$itests" --argjson u "$utests" --argjson r "$rest" --argjson o "$otests" '$i + $u + [$r] + $o')"
-
-          # Apply custom runner labels to the groups
-          groups="$(jq -n --argjson g "$groups" --argjson r "$runners" '$g | map(. + {"runner": (.name as $n | $r | .[$n]) })')"
-
-          # Apply the needs_parameters flag to the groups
-          groups="$(jq -n --argjson g "$groups" --argjson p "$parameters" '$g | map(. + {"needs_parameters": ([.name] | inside($p)) })')"
-
-          # Output the groups
-          echo "groups=$groups"
-          echo "groups=$(jq -nc --argjson g "$groups" '$g')" >> $GITHUB_OUTPUT
+          echo "groups<<EOF" >> $GITHUB_OUTPUT
+          go run ./cmd/ci/main.go --json list-test-groups | jq -r '.msg' | tee -a $GITHUB_OUTPUT
```

Comment:
Is the data model still the same?

My understanding of before:

```json
[
  { "name": "group1", "runner": "ubuntu-latest" },
  { "name": "group2", "runner": "ubuntu-24.04-arm" }
]
```

now:
```json
{
    "name": "itest-direct_data_onboard_verified",
    "runner": [
      "ubuntu-latest"
    ]
  }
```

Basically `runner` is an array now.  (Given its an array, lets change to runners to be clear?)

Reply:
Runner has always been an array. Before, it was also:

```
[
  { "name": "group1", "runner": ["ubuntu-latest"] },
  { "name": "group2", "runner": ["ubuntu-24.04-arm"] }
]
```

We needed them to be arrays because self-hosted runners have more than one label.

Reply:
Ack - ok.  Good to know.  My bad.  Resolving.